src/toolkits/image_deep_feature_extractor/image_deep_feature_extractor_toolkit.hpp

```diff
@@ -26,6 +26,7 @@ class EXPORT image_deep_feature_extractor_toolkit : public ml_model_base {
   void init_options(const std::map<std::string, flexible_type>& options) override;
 
   gl_sarray extract_features(gl_sframe data, const std::string& column_name, bool verbose, size_t batch_size) const;
+  gl_sarray sarray_extract_features(gl_sarray data, bool verbose, size_t batch_size) const;
```

Comment:
So I attempted to do that and I got some errors compiling due to something in the Registration Mechanism. Is there any reason in your mind this could happen?

Reply:
Yeah, there is some weirdness with the registration mechanism and overloads. I think it's theoretically possible to get it to work but I'm not sure how. For now, it's fine as is.

Reply:
Note that the registration macros don't incorporate any notion of type and are therefore brittle in the face of overloading.

Reply:
Hmm, any reason not just to converge to SArray version?

Reply:
I think an SArray version makes sense.  I think the argument against it is to be consistent with other feature extractors (only boosted trees ATM) that take multiple input columns.   However, this is pretty weak as that doesn't take the column name as part of it.